### PR TITLE
Add more variants of foreach usage

### PR DESCRIPTION
**Pull request by nordlow**

### CODE REVIEW

File: ctod.dd [hunk 1]
```diff
@@ -311,6 +311,21 @@ foreach (int value; array)
...
 
+In D the element type can be inferred even without using an explicit type qualifier:
+
+----------------------------
+int[17] array;
+foreach (value; array)
+    func(value);
+----------------------------
+
+or the through the ref keyword (for reference access):
```
Comment (CyberShadow) [inline ctod.dd]: the through the

File: ctod.dd [hunk 2]
```diff
@@ -311,6 +311,21 @@ foreach (int value; array)
     func(value);
 ----------------------------
 
+In D the element type can be inferred even without using an explicit type qualifier:
```
Comment (schuetzm) [inline ctod.dd]: comma after "In D"

**Comment by schuetzm** [comment]
Please squash, the LGTM

**Comment by nordlow** [comment]
Can I squash through the the browser interface?

File: ctod.dd [hunk 3]
```diff
@@ -311,6 +311,21 @@ foreach (int value; array)
...
+foreach (value; array)
+    func(value);
+----------------------------
+
+or through the ref keyword (for reference access):
+
+----------------------------
+int[17] array;
+foreach (ref value; array)
+    value = value + 42;
```
Comment (JakobOvrum) [inline ctod.dd]: `value += 42;`

**Comment by schuetzm** [comment]
Unfortunately not, you need to do it on your computer. See here for some nice explanations:

https://davidwalsh.name/squash-commits-git
https://ariejan.net/2011/07/05/git-squash-your-latests-commits-into-one/

In your case, you can use:
```bash
git checkout patch-2
git rebase -i master
git push -f
```

**Pull request merged**